csharp/src/Drivers/Apache/Impala/ImpalaConnection.cs

```diff
@@ -17,65 +17,572 @@
+                    string tableName = tableList[i];
+                    string columnName = columnNameList[i];
+                    short colType = (short)columnTypeList[i];
+                    string typeName = typeNameList[i];
+                    short nullable = (short)nullableList[i];
+                    string? isAutoIncrementString = isAutoIncrementList[i];
+                    bool isAutoIncrement = (!string.IsNullOrEmpty(isAutoIncrementString) && (isAutoIncrementString.Equals("YES", StringComparison.InvariantCultureIgnoreCase) || isAutoIncrementString.Equals("TRUE", StringComparison.InvariantCultureIgnoreCase)));
+                    string isNullable = isNullableList[i] ?? "YES";
+                    string columnDefault = columnDefaultList[i] ?? "";
+                    // Spark/Databricks reports ordinal index zero-indexed, instead of one-indexed
+                    int ordinalPos = ordinalPosList[i] + 1;
+                    TableInfo? tableInfo = catalogMap.GetValueOrDefault(catalog)?.GetValueOrDefault(schemaDb)?.GetValueOrDefault(tableName);
+                    tableInfo?.ColumnName.Add(columnName);
+                    tableInfo?.ColType.Add(colType);
+                    tableInfo?.Nullable.Add(nullable);
+                    tableInfo?.IsAutoIncrement.Add(isAutoIncrement);
+                    tableInfo?.IsNullable.Add(isNullable);
+                    tableInfo?.ColumnDefault.Add(columnDefault);
+                    tableInfo?.OrdinalPosition.Add(ordinalPos);
+                    SetPrecisionScaleAndTypeName(colType, typeName, tableInfo);
+                }
+            }
+
+            StringArray.Builder catalogNameBuilder = new StringArray.Builder();
+            List<IArrowArray?> catalogDbSchemasValues = new List<IArrowArray?>();
+
+            foreach (KeyValuePair<string, Dictionary<string, Dictionary<string, TableInfo>>> catalogEntry in catalogMap)
+            {
+                catalogNameBuilder.Append(catalogEntry.Key);
+
+                if (depth == GetObjectsDepth.Catalogs)
+                {
+                    catalogDbSchemasValues.Add(null);
+                }
+                else
+                {
+                    catalogDbSchemasValues.Add(GetDbSchemas(
+                                depth, catalogEntry.Value));
+                }
+            }
+
+            Schema schema = StandardSchemas.GetObjectsSchema;
+            IReadOnlyList<IArrowArray> dataArrays = schema.Validate(
+                new List<IArrowArray>
+                {
+                    catalogNameBuilder.Build(),
+                    catalogDbSchemasValues.BuildListArrayForType(new StructType(StandardSchemas.DbSchemaSchema)),
+                });
+
+            return new ImpalaInfoArrowStream(schema, dataArrays);
         }
 
-        protected override Task<TProtocol> CreateProtocolAsync(TTransport transport)
+        private static IReadOnlyDictionary<string, int> GetColumnIndexMap(List<TColumnDesc> columns) => columns
```

Comment:
Can move to parent.

Reply:
The implementation is different for GetColumnIndexMap.